agent/utils/fips_linux_test.go

```diff
@@ -0,0 +1,58 @@
+//go:build !windows && unit
+// +build !windows,unit
+
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package utils
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// detectFIPSModeWithPath is a helper function for testing
+func detectFIPSModeWithPath(filePath string) bool {
+	data, err := ioutil.ReadFile(filePath)
+	if err == nil && strings.TrimSpace(string(data)) == "1" {
+		return true
+	}
+	return false
+}
+func TestDetectFIPSMode(t *testing.T) {
```

Comment:
This unit test appears to just be testing detectFIPSModeWithPath instead of DetectFIPSMode so it's somewhat pointless.

Could we maybe make the DetectFIPSMode function take in the path as an argument, and make `fipsModeFilePath` an exported constant?

That way we can unit-test the actual DetectFIPSMode function by passing in the path, and the only downside is the caller will be slightly redundant like `utils.DetectFIPSMode(utils.FIPSModeFilePath)`

Reply:
Addressed in https://github.com/aws/amazon-ecs-agent/pull/4189/commits/421f95a1f0809cc35d3fcfbc4e8ac7567b8aa762